### PR TITLE
PIM-10683: Fix performance regression in CSV and XLSX product and pro…

### PR DESCRIPTION
**Pull request by ahocquard**
…duct model export

**Description (for Contributor and Core Developer)**

A cache query is not used inside the repository. It means that for each value, it triggers a SQL query. It slows down the CSV  and XLSX export probably by a factor between x2 and x5 depending of the catalog (tested on real catalog, I did a x3).

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php [hunk 1]
```diff
@@ -55,7 +46,7 @@ public function process($product): array
         $parameters = $this->stepExecution->getJobParameters();
         $structure = $parameters->get('filters')['structure'];
         $channel = $this->channelRepository->findOneByIdentifier($structure['scope']);
-
+        $jobLocales = $this->stepExecution->getJobParameters()->get('filters')['structure']['locales'];
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Processor/Normalization/ProductProcessor.php:49]: Can be moved to filterLocaleSpecificAttributeCodes directly

File: tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php [hunk 2]
```diff
@@ -268,4 +275,19 @@ public function it_processes_product_with_filter_on_quality_score(
 
         $this->process($product)->shouldBeLike($normalizedProductWithQualityScores);
     }
+
+    private function createAttribute(string $code): Attribute {
+        return new Attribute(
+            '$code',
```
Comment (StevenVAIDIE) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Connector/Processor/Normalization/ProductProcessorSpec.php:281]:
```suggestion
            $code,
```

**Review by StevenVAIDIE — APPROVED** [review]
Should we add a Spec ? For me it's should not be tested with phpspec but in case of you forget to add it

**Comment by ahocquard** [comment]
@StevenVAIDIE it's an implementation change already covered by some tests (integration test)

**Review by ValentinMumble — APPROVED** [review]

**Pull request merged**